src/common/AndroidUtils.ts

```diff
@@ -589,18 +588,44 @@ export class AndroidSDKUtils {
         return `'${pathName}'`;
     }
 
-    private static enableHWKeyboardForEmulator(
+    private static updateEmulatorConfig(
         emulatorName: string
     ): Promise<boolean> {
-        return new Promise((resolve, reject) => {
-            try {
-                shell
-                    .echo('hw.keyboard=yes')
-                    .toEnd(`~/.android/avd/${emulatorName}.avd/config.ini`);
+        return new Promise(async (resolve, reject) => {
+            const config = await AndroidSDKUtils.readEmulatorConfig(
+                emulatorName
+            );
+            if (config.size === 0) {
+                // If we cannot edit the AVD config, fail silently.
+                // This will be a degraded experience but should still work.
```

Comment:
nit: log some message stating the issue before failing silently

Reply:
Good call.